worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -769,12 +769,12 @@ class PartitionFilesCleaner {
             () -> {
               try {
                 while (!partitionFilesSorter.isShutdown()) {
+                  lock.lockInterruptibly();
                   try {
-                    lock.lockInterruptibly();
-                    while (queue.isEmpty()) {
-                      notEmpty.await(500, TimeUnit.MILLISECONDS);
+                    if (fileSorters.isEmpty()) {
+                      notEmpty.await();
```

Comment:
nit: change the `if` to `while` - in case of spurious wakeup's.
Note, the logic is pretty much the same in both cases (as iterator is empty even with spurious wakeups) - just that we will have a bit more lock/unlocks

Reply:
Updated.

@mridulm thanks for checking and suggestions, good luck in 2024 :)